### ISSUE TITLE
Self-attention in SMPL decoder is a no-op due to single query token

### ISSUE THREAD

**Issue by leonardwetzel** [pixelite1201/CameraHMR#26]
In `core/models/components/pose_transformer.py`, the `Attention` module is invoked with sequence length 1 (the single zero token from `smpl_head.py`). With `n=1`, `softmax(q·kᵀ)` is identically 1 regardless of `q` and `k`, so the self-attention reduces to:

```
to_out(v_chunk(LN(x))) + x
```

The Q and K projections inside `to_qkv` (≈ 2/3 of its parameters) receive zero gradient and remain at their init values. With the standard config (`dim=1024, heads=8, dim_head=64, depth=6`), this is roughly **6.3M dead parameters** plus their flops, repeated across all decoder layers in `TransformerCrossAttn`.

Quick check:

```python
# x shape: (B, 1, 1024) — single query token
qkv = to_qkv(x).chunk(3, dim=-1)        # q, k, v each (B, 1, 512)
dots = q @ k.transpose(-1, -2)           # (B, h, 1, 1)
attn = softmax(dots, dim=-1)             # always 1.0
out = attn @ v                           # == v
```

This appears to be inherited from HMR2.0's "standard transformer decoder" template, which assumed multiple query tokens but [here](https://github.com/shubham-goel/4D-Humans/blob/main/hmr2/models/heads/smpl_head.py) they set `token = torch.zeros(B, 1, 1)`.

Possible fixes, depending on intent:

1. **Inference-only:** skip the self-attention sub-block (or fold dead params away) — bit-identical outputs, free flops.
2. **Cleanup + retrain:** replace self-attn with `nn.Identity()` or a single `LN → Linear(dim, dim)`.
3. **Architectural fix:** I think architectures like SAM 3D Body use more learnable query tokens. Introducing tokens for pose, shape and maybe camera or keypoints can make the Self-Attention still useful. Do you think this can improve the model's performance?

Happy to open a PR for either solution if that's useful! Interested in what you say, maybe I am overseeing something.
